pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -385,6 +385,13 @@ public static class Server {
         "pinot.server.starter.enableRealtimeOffsetBasedConsumptionStatusChecker";
     public static final boolean DEFAULT_ENABLE_REALTIME_OFFSET_BASED_CONSUMPTION_STATUS_CHECKER = false;
 
+    public static final String CONFIG_OF_ENABLE_REALTIME_FRESHNESS_BASED_CONSUMPTION_STATUS_CHECKER =
+        "pinot.server.starter.enableRealtimeFreshnessBasedConsumptionStatusChecker";
+    public static final boolean DEFAULT_ENABLE_REALTIME_FRESHNESS_BASED_CONSUMPTION_STATUS_CHECKER = false;
+    public static final String CONFIG_OF_STARTUP_REALTIME_MIN_FRESHNESS_MS =
+        "pinot.server.starter.realtimeMinFreshnessMs";
+    public static final int DEFAULT_STARTUP_REALTIME_MIN_FRESHNESS_MS = 0;
```

Comment:
For a stream with high throughput, can this make it never able to catch up? I think a higher default is better, e.g. 10s

Reply:
the default of 0 is actually "off". but given the feedback above, i've set this to 10s, and this will get used if a user misconfigures this